kernel/tests/write.rs

```diff
@@ -251,3 +270,356 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
+fn set_value(
+    mut value: &mut serde_json::Value,
+    path: &str,
+    new_value: serde_json::Value,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let parts: Vec<_> = path.split('.').collect();
+
+    for name in parts.iter().take(parts.len() - 1) {
+        value = value
+            .get_mut(*name)
+            .ok_or_else(|| format!("key '{name}' not found"))?;
+    }
+
+    let last_key = parts.last().ok_or("empty path")?;
+    value
+        .as_object_mut()
+        .ok_or("expected a JSON object")?
+        .insert(last_key.to_string(), new_value);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_append() -> Result<(), Box<dyn std::error::Error>> {
+    // setup tracing
+    let _ = tracing_subscriber::fmt::try_init();
+    // setup in-memory object store and default engine
+    let (store, engine, table_location) = setup("test_table", true);
+
+    // create a simple table: one int column named 'number'
+    let schema = Arc::new(StructType::new(vec![StructField::new(
+        "number",
+        DataType::INTEGER,
+        true,
+    )]));
+    let table = create_table(store.clone(), table_location, schema.clone(), &[]).await?;
+
+    let commit_info = new_commit_info()?;
+
+    let mut txn = table
+        .new_transaction(&engine)?
+        .with_commit_info(commit_info);
+
+    // create two new arrow record batches to append
+    let append_data = [[1, 2, 3], [4, 5, 6]].map(|data| -> DeltaResult<_> {
+        let data = RecordBatch::try_new(
+            Arc::new(schema.as_ref().try_into()?),
+            vec![Arc::new(arrow::array::Int32Array::from(data.to_vec()))],
+        )?;
+        Ok(Box::new(ArrowEngineData::new(data)))
+    });
+
+    // write data out by spawning async tasks to simulate executors
+    let engine = Arc::new(engine);
+    let write_context = Arc::new(txn.get_write_context());
+    let tasks = append_data.into_iter().map(|data| {
+        // arc clones
+        let engine = engine.clone();
+        let write_context = write_context.clone();
+        tokio::task::spawn(async move {
+            engine
+                .write_parquet(
+                    data.as_ref().unwrap(),
+                    write_context.as_ref(),
+                    HashMap::new(),
+                    true,
+                )
+                .await
+        })
+    });
+
+    let write_metadata = futures::future::join_all(tasks).await.into_iter().flatten();
+    for meta in write_metadata {
+        txn.add_write_metadata(meta?);
+    }
+
+    // commit!
+    txn.commit(engine.as_ref())?;
+
+    let commit1 = store
+        .get(&Path::from(
+            "/test_table/_delta_log/00000000000000000001.json",
+        ))
+        .await?;
+
+    let mut parsed_commits: Vec<_> = Deserializer::from_slice(&commit1.bytes().await?)
+        .into_iter::<serde_json::Value>()
+        .try_collect()?;
+
+    set_value(&mut parsed_commits[0], "commitInfo.timestamp", json!(0))?;
```

Comment:
can you note that you're setting things here so the comparison works?

also, does the modification time stuff work on the in-mem store? If so, it might be worth checking that it's within a couple of seconds of "now" to validate that it's properly propagating.

Same in other test.

Reply:
this caught a bug thanks nick :)